rsocket/RSocketRequester.cpp

```diff
@@ -17,14 +17,18 @@ namespace rsocket {
 RSocketRequester::RSocketRequester(
     std::shared_ptr<RSocketStateMachine> srs,
     EventBase& eventBase)
-    : stateMachine_(std::move(srs)), eventBase_(eventBase) {}
+    : stateMachine_(std::move(srs)), eventBase_(&eventBase) {}
 
 RSocketRequester::~RSocketRequester() {
   VLOG(1) << "Destroying RSocketRequester";
 }
 
+void RSocketRequester::setEventBase(folly::EventBase* evb) {
+  eventBase_ = evb;
```

Comment:
There is no way this can be safe, if there are pending callbacks on the old evb you'll now have interleaved callbacks from different threads. This needs to use the EventBase switcher that @dymk (I think?) worked on. Basically enqueuing the change on the original evb and then collecting callbacks until that executes. See https://github.com/rsocket/rsocket-cpp/blob/master/rsocket/internal/SwappableEventBase.h

Reply:
Yep, I think this would require rewriting RSocketRequester to use SwappableEventBase and then enqueuing the swap on that, to keep events in rsocket nice and serialized.

Reply:
The only case that we need event base switching is for Thrift integration. Thrift integration doesn't require having SwappableEventBase implementation. At least for the current state. 

Neither Proxygen have anything like SwappableEventBase. It is not a requirement for this feature.